### PR TITLE
Run dead-code:prod locally and handle findings

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Fixes the errors from running the new `npm run dead-code:prod`  command

## Discussion

- Most of the unused exports flagged in production mode were used internally to the file and only exported for testing
- A handful of files actually had dead code that was only kept around by tests -- see comments

## Future Work

- Enforce running in CI

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/starterBricks/starterBrickModUtils.test.ts
Comment (grahamlangford) [inline src/starterBricks/starterBrickModUtils.test.ts:1]: Dead code

File: src/starterBricks/starterBrickModUtils.ts
Comment (grahamlangford) [inline src/starterBricks/starterBrickModUtils.ts:1]: Dead code

File: src/telemetry/telemetryHelpers.test.ts
Comment (grahamlangford) [inline src/telemetry/telemetryHelpers.test.ts:1]: tests for dead code

File: src/utils/inference/markupInference.test.tsx
Comment (grahamlangford) [inline src/utils/inference/markupInference.test.tsx:1]: tests for dead code

File: src/utils/inference/markupInference.ts
Comment (grahamlangford) [inline src/utils/inference/markupInference.ts:1]: dead code

File: src/utils/inference/selectorInference.test.ts
Comment (grahamlangford) [inline src/utils/inference/selectorInference.test.ts:1]: tests for dead code

File: src/utils/modUtils.ts
Comment (grahamlangford) [inline src/utils/modUtils.ts:1]: dead code

File: src/utils/timeUtils.ts
Comment (grahamlangford) [inline src/utils/timeUtils.ts:1]: Only used in testing and we do have strictNullChecks, so removed the check

Comment (twschiller) [reply src/utils/timeUtils.ts:1]: Consider making a separate PR since it's not a backward compatible change

**Review by twschiller — APPROVED** [review]
https://github.com/pixiebrix/pixiebrix-extension/pull/9329/files#r1811318413

**Comment by twschiller** [comment]
Main considerations: https://github.com/pixiebrix/pixiebrix-extension/pull/9329/files#r1811318413

File: src/utils/timeUtils.ts
Comment (grahamlangford) [reply src/utils/timeUtils.ts:1]: What's the concern? It is only used by tests, so there is no risk that it's breaking anything

Comment (twschiller) [reply src/utils/timeUtils.ts:1]: > What's the concern? It is only used by tests, so there is no risk that it's breaking anything

Gotcha - I didn't realize the method was only used in test code. 👍 You're good to go

**Pull request merged**